lib/resty/upstream/healthcheck.lua

```diff
@@ -92,6 +93,92 @@ local function set_peer_down_globally(ctx, is_backup, id, value)
     end
 end
 
+local function get_peers_to_check(peers)
+    local peers_to_check = {}
+    for _, peer in ipairs(peers) do
+        if not checked_peers_upstream[peer.name] then
+            table.insert(peers_to_check, peer)
+        end
+    end
+    return peers_to_check
+end
+
+local function get_checked_peers(peers)
+    local checked_peers = {}
+    for _, peer in ipairs(peers) do
+        if checked_peers_upstream[peer.name] then
+            table.insert(checked_peers, peer)
+        end
+    end
+    return checked_peers
+end
+
+local function get_all_upstream_peers(u)
```

Comment:
Function names should be consistent. Currently, we have: `get_all_upstream_peers`, `get_unique_upstream_peers`, `already_checked_peers` returning a pair of arrays (primary & backup), and `get_peers_to_check` and `get_checked_peers` returning a single filtered array. `already_checked_peers` returns the result of a pair of calls to `get_checked_peers` and `get_unique_upstream_peers` returns the result of a pair of calls to `get_peers_to_check`.
The following renames would be clearer:
- `get_all_upstream_peers -> get_all_peers`
- `get_unique_upstream_peers -> get_all_unchecked_peers`
- `get_peers_to_check  -> get_unchecked_peers`
- `already_checked_peers -> get_all_checked_peers`